packages/cli-kit/src/public/node/ruby.test.ts

```diff
@@ -182,3 +183,75 @@ describe('execCLI', () => {
     process.env = originalEnv
   })
 })
+
+it('when it run with spin then the it passes spin configuration to the CLI2', async () => {
+  // Setup
+  const originalEnv = process.env
+
+  // Given
+  const execSpy = vi.spyOn(system, 'exec')
```

Comment:
ok, I have followed how the previous tests where made. Anyway I don't understand exactly  what you mean with this, in case I want to mock only some parts of the module, I think that it's better to use `spyOn` right?, so the rest of the methods of that module are not mocked

Reply:
`vi.mock`: Mock all the exported elements from the module.
`vi.spyOn`: Use the actual module and observe calls.

By defaulting to the latter, we are defaulting to testing the integration of the subject under test with the modules it depends on. That might be what we want, but my recommendation is that we default to our test being a "unit test", or in other words everything the module depends on is mocked, and then purposely remove mocks.

Reply:
ok, however in this case with `execSpy` we need to `spy` on it to verify that the correct parameters are used when the `exec` method is called